src/foraging_gui/MyFunctions.py

```diff
@@ -890,7 +897,56 @@ def _LickSta(self,Trials=None):
             self.Start_CoCue_LeftRightRatio=np.nan
         else:
             self.Start_CoCue_LeftRightRatio=np.array(sum(self.Start_GoCue_LeftLicks))/np.array(sum(self.Start_GoCue_RightLicks))
-        
+
+    def calculate_inter_lick_intervals(self):
+        """
+        Calculate and categorize lick intervals
+        """
+
+        right = self.B_RightLickTime
+        left = self.B_LeftLickTime
+        threshold = .1
+
+        if right is not None and left is not None and len(right) > 0 and len(left) > 0:
+            # calculate left interval and fraction
+            same_side_l = np.diff(left)
+            same_side_l_frac = len(same_side_l[~(same_side_l > threshold)]) / len(same_side_l)
+            logging.info(f'Percentage of left lick intervals under 100 ms is {same_side_l_frac * 100}%.')
+            self.B_LeftLickIntervalPercent = same_side_l_frac * 100
+
+            # calculate right interval and fraction
+            same_side_r = np.diff(right)
+            same_side_r_frac = len(same_side_r[~(same_side_r > threshold)]) / len(same_side_r)
+            logging.info(f'Percentage of right lick intervals under 100 ms is {same_side_r_frac * 100}%.')
+            self.B_RightLickIntervalPercent = same_side_r_frac * 100
+
+            if same_side_l_frac + same_side_r_frac >= .1:
```

Comment:
I'm not sure if adding these fractions is what we want. 
I think if either the left or right is above .1 (which you should set as a variable instead of hard coding), then we should alert the user

Reply:
Okay I was doing that based on what you said [here](https://github.com/orgs/AllenNeuralDynamics/projects/47/views/10?sortedBy%5Bdirection%5D=asc&sortedBy%5BcolumnId%5D=Assignees&sortedBy%5Bdirection%5D=desc&sortedBy%5BcolumnId%5D=Status&pane=issue&itemId=69427925&issue=AllenNeuralDynamics%7Caind-behavior-blog%7C308#:~:text=If%20the%20proportion%20of%20intervals%20is%20%3E10%25%20for%20same%20side%2C%20show%20a%20message.%20(That%20is%2C%2010%25%20of%20all%20same%2Dside%20intervals%2C%20combining%20across%20left%2Dleft%20and%20right%2Dright)). I will change to if either is over the threshold

Reply:
I suppose its hard to implement things when I tell you inconsistent things. Sorry!

I'm not sure if we want to check if either left or right same side, or combine across sides. But if we do combine, then instead of adding the fractions you should instead concatenate the array of diffs and then compute the fraction:

```
same_side_l = np.diff(left)
same_side_r = np.diff(right)
same_side_combined = np.concat([same_side_l, same_side_r])
same_side_frac = np.mean(same_side_combined <= threshold)
```

If we add the fractions, then the combined fraction ranges from 0 to 2, so that doesn't make sense. 

I guess I would err on the side of checking the combined fraction, since it should be more stable when there are a small number of licks